### ISSUE TITLE
'vagrant up' fails with '`windows_uninherit_handle': The parameter is incorrect. - SetHandleInformation (SystemCallError)'

### ISSUE THREAD

**Issue by RickJWagner** [dustymabe/vagrant-sshfs#52]
A user is reporting this issue:

```
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
==> default: Registering box with vagrant-registration...
==> default: Checking for guest additions in VM...
    default: No guest additions were detected on the base box for this VM! Guest
    default: additions are required for forwarded ports, shared folders, host only
    default: networking, and more. If SSH fails on this machine, please install
    default: the guest additions and repackage the box to continue.
    default:
    default: This is not an error message; everything may continue to work properly,
    default: in which case you may ignore this message.
==> default: Configuring and enabling network interfaces...
==> default: Configuring proxy for Docker...
==> default: Configuring proxy environment variables...
==> default: Configuring proxy for Yum...
==> default: Copying TLS certificates to C:/cdk/cdk/components/rhel/rhel-ose/.vagrant/machines/default/virtualbox/docker
==> default: Mounting SSHFS shared folder...
==> default: Mounting folder via SSHFS: C:/Users/ianp => /c/Users/ianp
C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/cap/guest/linux/sshfs_forward_mount.rb:152:in `windows_uninherit_handle': The parameter is incorrect. - SetHandleInformation (SystemCallError)
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/cap/guest/linux/sshfs_forward_mount.rb:122:in `block in windows_uninherit_handles'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/cap/guest/linux/sshfs_forward_mount.rb:118:in `each_object'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/cap/guest/linux/sshfs_forward_mount.rb:118:in `windows_uninherit_handles'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/cap/guest/linux/sshfs_forward_mount.rb:226:in `sshfs_slave_mount'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/cap/guest/linux/sshfs_forward_mount.rb:75:in `sshfs_forward_mount_folder'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/capability_host.rb:111:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/capability_host.rb:111:in `capability'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/guest.rb:43:in `capability'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/synced_folder/sshfs_forward_mount.rb:52:in `do_forward_mount'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/synced_folder.rb:47:in `block in enable'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/synced_folder.rb:42:in `each'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-sshfs-1.2.0/lib/vagrant-sshfs/synced_folder.rb:42:in `enable'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/synced_folders.rb:93:in `block in call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/synced_folders.rb:90:in `each'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/synced_folders.rb:90:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/synced_folder_cleanup.rb:28:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/synced_folders/nfs/action_cleanup.rb:19:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/prepare_nfs_valid_ids.rb:12:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/handle_forwarded_port_collisions.rb:49:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/prepare_forwarded_port_collision_params.rb:30:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/env_set.rb:19:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-proxyconf-1.5.2/lib/vagrant-proxyconf/action/only_once.rb:21:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/ianp/.vagrant.d/gems/gems/vagrant-service-manager-1.3.3/lib/vagrant-service-manager/action/log_configured_service.rb:13:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:80:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/clear_forwarded_ports.rb:15:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/set_name.rb:19:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/clean_machine_folder.rb:17:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_accessible.rb:18:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/box_check_outdated.rb:36:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:224:in `action_raw'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:199:in `block in action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/environment.rb:561:in `lock'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `call'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `action'
        from C:/cdk/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

**Comment by RickJWagner** [comment]
Asking the user for more environmental information now.

**Comment by dustymabe** [comment]
hey @RickJWagner - any more info from the user? Would like to be able to recreate this so that I can better understand the problem and also verify it is fixed when I make changes to the code.

**Comment by RickJWagner** [comment]
Hi Dusty,
Nothing yet.  I've left myself a note on the support case, when the user tells us more I'll update this tracker.

**Comment by Joseph-Watkins** [comment]
Hi, I've just had exactly the same issue. Windows 7 64 bit. What other env info do you need?

**Comment by dustymabe** [comment]
hey @Joseph-Watkins. Sorry I didn't respond sooner. I was traveling.

Any information would help. What Vagrantfile are you using? What shell are you using (cygwin, powershell, etc..)? Can you reproduce this reliably? Can you try to help me reproduce?

**Comment by Joseph-Watkins** [comment]
Ah, np. I'm using the Red Hat cdk 2.2 download (http://developers.redhat.com/products/cdk/download/ ) and the vagrant file included with that. Have tried from cygwin and windows cmd shell, with same result. Happens every time. The zip install includes vagrant-sshfs-1.2.0.gem

**Comment by dustymabe** [comment]
hey @Joseph-Watkins can you please try the following and tell me if it works for you now:

- download the file at https://dustymabe.fedorapeople.org/vagrant-sshfs-1.2.1.a7d8a89.gem
- `vagrant plugin uninstall vagrant-sshfs`
- `vagrant plugin install ./vagrant-sshfs-1.2.1.a7d8a89.gem`
- test to see if your problem is fixed

**Comment by Joseph-Watkins** [comment]
Hi, thanks. no not fixed, but maybe more info, now get:

==> default: Configuring and enabling network interfaces...
==> default: Copying TLS certificates to C:/Joe/Installs/Work/RedHat_cdk/components/rhel/rhel-ose/.vagrant/machines/defau
lt/virtualbox/docker
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!
readlink -f /c/Users/joe
Stdout from the command:
Stderr from the command:


The directory /c/Users/joe doesn't exist, assume should be /cygdrive/c/Users/joe, but not sure where this is coming from?

**Comment by dustymabe** [comment]
hey @Joseph-Watkins turns out there was an issue I introduced in a recent commit that was unrelated to this fix. Try out this new dev version please:


- download the file at https://dustymabe.fedorapeople.org/vagrant-sshfs-1.2.1.b3eae35.gem
- `vagrant plugin uninstall vagrant-sshfs`
- `vagrant plugin install ./vagrant-sshfs-1.2.1.b3eae35.gem`
- test to see if your problem is fixed

**Comment by Joseph-Watkins** [comment]
Hi, yes, that seems to have fixed it - thanks!

**Comment by dustymabe** [comment]
Fixed in fe4fcd5 that was a part of #57

**Comment by dustymabe** [comment]
Note that the fix here is to not error, but warn, if there is an error when setting a file handle to uninheritable. Hopefully this will take care of the problem, but if it does not we will have more verbose output that can be used to diagnose.
